### PR TITLE
Update Java style guide (January 2022 revisions)

### PR DESCRIPTION
**Pull request by alexbishop1**
The Java community meeting reviewed the [Java style guide](https://gds-way.cloudapps.digital/manuals/programming-languages/java.html) on 14 January 2022.

There was rough consensus on some changes, which are detailed in the individual commit messages.

### CODE REVIEW

File: source/manuals/programming-languages/java.html.md.erb [hunk 1]
```diff
@@ -112,14 +112,16 @@ The [OpenJDK project has some style guidelines for local variable type inference
...
 
-When using external libraries, favour those which complement the Java standard library. For example, be wary of any library which introduces a new type that replicates the functionality of an existing type in the Java standard library without implementing the same interfaces (for example, a list type which does not implement `java.util.List`).
+When using external libraries, favour those which complement the Java standard library. For example, consider a library that introduces an object that behaves like a list. If this type does not implement `java.util.List`, it will not automatically benefit from the standard library’s support for features like streams.
+
+be wary of any library which introduces a new type that replicates the functionality of an existing type in the Java standard library without implementing the same interfaces (for example, a list type which does not implement `java.util.List`).
```
Comment (philandstuff) [inline source/manuals/programming-languages/java.html.md.erb:119]: This line looks like it was unintentionally left in?

**Review by philandstuff — COMMENTED** [review]
Thanks for this! One weird bit but otherwise looks great 👍🏻

File: source/manuals/programming-languages/java.html.md.erb [hunk 2]
```diff
@@ -137,7 +137,7 @@ We encourage the use of static analysis. Static analysis tools for Java include
 
 If possible, configure your static analysis tools using configuration files and keep these in your project repositories. This makes your settings more portable and makes it easier to perform the same checks in different places (for example, in a cloud service and on your own computer).
 
-Some cloud-based static analysis tools, including Codacy, can work directly with GitHub repositories. Only grant a tool access to your repositories if it has been approved by an information assurance review.
+Some cloud-based static analysis tools, including Codacy, can work directly with GitHub repositories. Only grant a tool access to your repositories if it has been approved by an information assurance review. Some teams are using [SonarCloud](https://sonarcloud.io/), which has been approved.
```
Comment (mrwilson) [inline source/manuals/programming-languages/java.html.md.erb:142]: It has been approved specifically for public repos only, which SonarCloud provides for free (at least, within Digital Identity)

**Comment by mrwilson** [comment]
Looks good, thanks Alex! 🎉

File: source/manuals/programming-languages/java.html.md.erb (hunk 1 above)
Comment (alexbishop1) [reply source/manuals/programming-languages/java.html.md.erb]: That’s an editing mistake. Fixed.

File: source/manuals/programming-languages/java.html.md.erb (hunk 2 above)
Comment (alexbishop1) [reply source/manuals/programming-languages/java.html.md.erb]: Rewritten that sentence to make the scope of SonarCloud approval clearer.

**Review by philandstuff — APPROVED** [review]

**Review by mrwilson — APPROVED** [review]

File: source/manuals/programming-languages/java.html.md.erb (hunk 2 above)
Comment (mrwilson) [reply source/manuals/programming-languages/java.html.md.erb]: Thanks!

**Pull request merged**